api/src/main/java/org/apache/iceberg/DataFile.java

```diff
@@ -98,18 +98,20 @@ public interface DataFile extends ContentFile<DataFile> {
   Types.NestedField SORT_ORDER_ID =
       optional(140, "sort_order_id", IntegerType.get(), "Sort order ID");
   Types.NestedField SPEC_ID = optional(141, "spec_id", IntegerType.get(), "Partition spec ID");
+  Types.NestedField SCHEMA_ID = optional(142, "schema_id", IntegerType.get(), "Schema ID");
 
   int PARTITION_ID = 102;
   String PARTITION_NAME = "partition";
   String PARTITION_DOC = "Partition data tuple, schema based on the partition spec";
-  // NEXT ID TO ASSIGN: 142
+  // NEXT ID TO ASSIGN: 143
 
   static StructType getType(StructType partitionType) {
     // IDs start at 100 to leave room for changes to ManifestEntry
     return StructType.of(
         CONTENT,
         FILE_PATH,
         FILE_FORMAT,
+        SCHEMA_ID,
```

Comment:
@szehon-ho if you are concerned about this order. We may put the `SCHEMA_ID` to last to align other new fields. Then we don't need to update too many methods of `get/put` in `BaseFile`.  And some of the spark UTs are not needed to update as well.

Reply:
Moved the last